File: Java/Migration/unit-tests-java/src/test/java/p01_core_runner/Ex04RepeatedTests.java
package p01_core_runner;

import org.testng.annotations.Test;

import java.util.concurrent.ThreadLocalRandom;

import static org.assertj.core.api.Assertions.assertThat;

public class Ex04RepeatedTests {
  private final ThreadLocalRandom current = ThreadLocalRandom.current();

  //non-deterministic test
  @Test//(invocationCount = 100) //running multiple times helps in diagnosis:
  public void shouldBeGreaterThan10() {
    //GIVEN
    int buggyVariable = current.nextInt(0, 100);

    //THEN
    assertThat(buggyVariable).isGreaterThan(10); //low chance of happening
  }
}

File: Java/Migration/unit-tests-csharp/unit-tests-csharp/P01CoreRunner/Ex04RepeatedTests.cs
using System;
using NUnit.Framework;

namespace unit_tests_csharp.P01CoreRunner
{
  public class Ex04RepeatedTests
  {
    private readonly Random _random = new Random(DateTime.UtcNow.Millisecond);

    //non-deterministic test
    [Test]//[Repeat(100)] //running multiple times helps in diagnosis:
    public void ShouldBeGreaterThan10()
    {
      //GIVEN
      var buggyVariable = _random.Next(0, 100);

      //THEN
      Assert.Greater(buggyVariable, 10); //low chance of happening
    }
  }
}
